doubly_linked_list/doubly_linked_list.py

```diff
@@ -1,68 +1,163 @@
+from math import inf
+
 """Each ListNode holds a reference to its previous node
 as well as its next node in the List."""
+
+
 class ListNode:
-  def __init__(self, value, prev=None, next=None):
-    self.value = value
-    self.prev = prev
-    self.next = next
+    def __init__(self, value, prev=None, next=None):
+        self.value = value
+        self.prev = prev
+        self.next = next
 
-  """Wrap the given value in a ListNode and insert it
+    """Wrap the given value in a ListNode and insert it
   after this node. Note that this node could already
   have a next node it is point to."""
-  def insert_after(self, value):
-    current_next = self.next
-    self.next = ListNode(value, self, current_next)
-    if current_next:
-      current_next.prev = self.next
 
-  """Wrap the given value in a ListNode and insert it
+    def insert_after(self, value):
```

Comment:
This is one of those mind benders that the robot_sort from the previous week definitely helps you pickup faster each time you got to contort the brain to work this out.